tests/run-make/reset-codegen-1/rmake.rs

```diff
@@ -0,0 +1,38 @@
+// When rustc received 4 codegen-units, an output path and an emit flag all simultaneously,
+// this could cause an annoying recompilation issue, uselessly lengthening the build process.
+// A fix was delivered, which resets codegen-units to 1 when necessary,
+// but as it directly affected the way codegen-units are manipulated,
+// this test was created to check that this fix did not cause compilation failures.
+// See https://github.com/rust-lang/rust/issues/30063
+
+//@ ignore-cross-compile
+
+use run_make_support::{rustc, tmp_dir};
+use std::fs;
+
+fn compile(emit: Option<&str>, output_dir: &str) {
+    let mut rustc = rustc();
+    let rustc = rustc.codegen_units(4).output(tmp_dir().join(output_dir)).input("foo.rs");
```

Comment:
Question: is this `output_dir` a directory or a file?

Reply:
You're right, it's a file, specified by the `-o` rustc flag and it's being removed individually by `remove_file`. Renamed.